components/dialogs/dialogs.js

```diff
@@ -37,15 +38,24 @@ components.register('TooManyFilesDialogViewModel', function(args) {
 });
 
 function DialogViewModel(title) {
-  this.closed = new signals.Signal();
+  var self = this;
+  this.onclose = null;
   this.title = ko.observable(title);
   this.taDialogName = ko.observable('');
+  this.closePromise = new Promise(function(resolve) {
+    self.onclose = resolve;
+  });
+}
+DialogViewModel.prototype.closeThen = function(thenFunc) {
+  this.closePromise = this.closePromise.then(thenFunc);
+  return this;
 }
 DialogViewModel.prototype.setCloser = function(closer) {
   this.close = closer;
 }
-DialogViewModel.prototype.onclose = function() {
-  this.closed.dispatch();
+DialogViewModel.prototype.publish = function() {
```

Comment:
s/publish/show/

So I'm not a big fan of object context changing in middle of function chain, that sounds little confusing.  With this suggestion, function chains switch it's context to promise from dialog object after publish is called.  If we were to chain functions it's context should be consistent through out entire chain.